member-service/src/main/java/com/hedvig/memberservice/web/InternalMembersController.java

```diff
@@ -179,5 +174,11 @@ public void editMemberInfo(
         //return ResponseEntity.notFound().build();
 
     }
+
+    @DeleteMapping("/{memberId}")
+    public ResponseEntity<Boolean> deleteMember(@PathVariable Long memberId) {
+        commandBus.sendAndWait(new DeleteMemberCommand(memberId));
+        return ResponseEntity.ok(true);
+    }
```

Comment:
This would simply print the string `"true"` as the response body. Doesn't it make more sense to simply return an empty 200 for success?

```suggestion
    @DeleteMapping("/{memberId}")
    public ResponseEntity<Void> deleteMember(@PathVariable Long memberId) {
        commandBus.sendAndWait(new DeleteMemberCommand(memberId));
        return ResponseEntity.ok().build();
    }
```

Reply:
Well, that's a subtle but interesting point you're making actually. I have bumped into this "issue" many times before myself when designing REST APIs. The short answer is that returning an empty response body makes the response invalid JSON. So then you need to explicitly set "content-type: text/plain" for this end-point and all clients need to special case it. If you instead continue to use "content-type: application/json" like all the other end-points, then clients that have a strict JSON parser will fail to parse the response and consider the request failed, even if it actually succeeded.

In my mind there are only two sensible things to return in a case like that. Either an empty object `{}`, which would be the JSON equivalent of an empty response body. Or a boolean `true` or `false`, depending on if anything was deleted (the request can still `200` succeed, even if nothing was actually deleted). And out of those two options, I consider the second one to be the most sensible one for a `DELETE` request like this.

But my reasoning might be completely off on this, so please prove me wrong if you want! 🌝